lib/git-hooks.js

```diff
@@ -131,14 +146,18 @@ function runHooks(hooks, args, callback) {
         return;
     }
 
-    var hook = spawnHook(hooks.shift(), args);
-    hook.on('close', function (code) {
-        if (code === 0) {
-            runHooks(hooks, args, callback);
-        } else {
-            callback(code);
-        }
-    });
+    try {
```

Comment:
try-catch here is required because we cannot catch the error in the hook(from hook-template) because the code became asynchronous -  excludeIgnoredPaths is async now.
That's why I tried to use sync version of exec